### PR TITLE
ADR-68: Colocating managed services on the same data plane OSD fleet

### PR DESCRIPTION
**Pull request by tombentley**

### CODE REVIEW

File: _adr/68/index.adoc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+## Proposed Architecture
+
+### Capacity Isolation
+
+A new machine pool will be created for each new managed service. A machine pool is a group of worker nodes in a cluster that have the same configuration, providing ease of management.
+
+This allows each new managed service to determine their own instance type that suits their needs. For example, M5 (general purpose), C5 (CPU optimised), R5 (Memory optimised) and at the size they would like.
+
+All machine pools will be created through the OpenShift Cluster Manager (OCM) API, not directly on the OpenShift Dedicated (OSD) clusters. Each machine pool will be created with a NoSchedule taint which will require all pods scheduled for that managed service to include the tolleration.
```
Comment (machi1990) [inline _adr/68/index.adoc:47]:
```suggestion
All machine pools will be created through the OpenShift Cluster Manager (OCM) API, not directly on the OpenShift Dedicated (OSD) clusters. Each machine pool will be created with a `NoSchedule` taint which will require all pods scheduled for that managed service to include the toleration.
```

File: _adr/68/index.adoc [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+### Capacity Isolation
+
+A new machine pool will be created for each new managed service. A machine pool is a group of worker nodes in a cluster that have the same configuration, providing ease of management.
+
+This allows each new managed service to determine their own instance type that suits their needs. For example, M5 (general purpose), C5 (CPU optimised), R5 (Memory optimised) and at the size they would like.
+
+All machine pools will be created through the OpenShift Cluster Manager (OCM) API, not directly on the OpenShift Dedicated (OSD) clusters. Each machine pool will be created with a `NoSchedule` taint which will require all pods scheduled for that managed service to include the toleration.
+
+While outside the scope of this ADR, it is worth noting that with a shared data plane fleet, a new micro service in the control plane would be ideal to control scaling machine pools, creation and deletion of data plane OSD clusters.
```
Comment (machi1990) [inline _adr/68/index.adoc:49]: I suppose this is talking about the [Terraforming API proposal](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/issues/4), correct? @davidffrench Maybe we can add a link.

File: _adr/68/index.adoc [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+* Scaling the nodes up and down becomes significantly more complex as the logic needs to consider all managed services.
+
+### Separate Data Plane 
+
+A separate data plane fleet for each managed service would require 6 additional nodes (3 master and 3 infra) for each OSD cluster. The significant cost impact here is for regions which are supported but rarely used across all managed services. With a separate data plane fleet, each new managed service would require at least one of their own OSD clusters in all supported regions.
+
+## Challenges
+
+Currently, all terraforming of a data plane RHOSAK cluster is performed through kas-fleet-manager. This logic should be split into its own microservice which will allow for supporting additional use cases such as the proposal here.
```
Comment (machi1990) [inline _adr/68/index.adoc:79]: same comment as https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/13/files#r830917388

**Comment by tombentley** [comment]
I'm closing this PR. I think it exists from a time early on in the move to using a github repo. The draft version of this ADR already exists, and we can sync with the gdoc content at the same time we move all the existing ADRs from gdoc.

**Pull request closed without merging**